src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/DocumentTemplateService.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.crowncommercial.dts.scale.cat.service;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import uk.gov.crowncommercial.dts.scale.cat.config.Constants;
+import uk.gov.crowncommercial.dts.scale.cat.exception.ResourceNotFoundException;
+import uk.gov.crowncommercial.dts.scale.cat.model.DocumentAttachment;
+import uk.gov.crowncommercial.dts.scale.cat.model.DocumentKey;
+import uk.gov.crowncommercial.dts.scale.cat.model.entity.ProcurementEvent;
+import uk.gov.crowncommercial.dts.scale.cat.model.generated.DocumentAudienceType;
+import uk.gov.crowncommercial.dts.scale.cat.model.generated.DocumentSummary;
+
+/**
+ *
+ */
+@Service
+@RequiredArgsConstructor
+public class DocumentTemplateService {
+
+  static final String FMT_TEMPLATE_DESCRIPTION = "Proforma Bid Pack Document (%s)";
+  static final String ERR_MSG_FMT_NO_TEMPLATES_FOR_EVENT_TYPE =
+      "No templates found for event type [%s]";
+  static final String ERR_MSG_FMT_TEMPLATE_NOT_FOUND =
+      "Template [ID: %s, Filename: %s] not found for event type [%s]";
+
+  // {project ID}-{event type}-{project name}.odt
+  static final String FILENAME_FMT_DRAFT_PROFORMA = "%d-%s-%s.odt";
+
+  private final ValidationService validationService;
+  private final DocumentTemplateSourceService documentTemplateSourceService;
+  private final DocGenService docGenService;
+
+  /**
+   * Lists the template documents for a given procurement event (by event type).
+   *
+   * @param procId
+   * @param eventId
+   * @return a collection of document summaries
+   */
+  public Collection<DocumentSummary> getTemplates(final Integer procId, final String eventId) {
+
+    var event = validationService.validateProjectAndEventIds(procId, eventId);
+    var resources = documentTemplateSourceService.getEventTypeTemplates(event.getEventType());
+
+    return resources.stream().map(r -> {
+      var docKey = new DocumentKey(Math.abs(r.getFilename().hashCode()), r.getFilename(),
```

Comment:
This is getting a Sonar msg - `Neither "Math.abs" nor negation should be used on numbers that could be "MIN_VALUE" (java:S2676)`

(just a minor one - happy to ignore)